src/linux_android.rs

```diff
@@ -15,15 +15,21 @@ pub fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
     static HAS_GETRANDOM: LazyBool = LazyBool::new();
     if HAS_GETRANDOM.unsync_init(is_getrandom_available) {
         sys_fill_exact(dest, |buf| unsafe {
-            libc::syscall(libc::SYS_getrandom, buf.as_mut_ptr(), buf.len(), 0) as libc::ssize_t
+            libc::syscall(libc::SYS_getrandom,
+                          buf.as_mut_ptr() as *mut libc::c_void,
+                          buf.len() as libc::size_t,
+                          0 as libc::c_uint) as libc::ssize_t
```

Comment:
Ah, you looked up the proper types, nice. :)
I am going to close my PR then.

I think it would be a good idea though to add a local helper method that takes arguments of the right types and just forwards them to the syscall. Then the types do not have to be duplicated.